### PR TITLE
community/ocaml: upgrade to 4.09.0

### PR DESCRIPTION
**Pull request by jjsearle**

### CODE REVIEW

File: community/ocaml/APKBUILD [hunk 1]
```diff
@@ -102,8 +96,6 @@ runtime() {
 		'*.cma' \
 		'VERSION' \
 		'stublibs' \
-		'vmthreads/*.cmi' \
-		'vmthreads/*.cma' \
```
Comment (jjsearle) [inline community/ocaml/APKBUILD]: The vmthreads library is no longer available. This library was deprecated in 4.08.0.

**Comment by maxice8** [comment]
note that this requires rebuilding the whole ocaml ecosystem

File: community/ocaml/APKBUILD [hunk 2]
```diff
@@ -18,37 +18,31 @@ subpackages="
 	$pkgname-compiler-libs:_compiler_libs
 	$pkgname-runtime
 	"
-source="http://caml.inria.fr/pub/distrib/ocaml-${pkgver%.*}/$pkgname-$pkgver.tar.gz
+source="http://caml.inria.fr/pub/distrib/ocaml-${pkgver%.*}/ocaml-$pkgver.tar.gz
```
Comment (XVilka) [inline community/ocaml/APKBUILD]: why did you remove `$pkgname`? Should be fine as is.

**Comment by XVilka** [comment]
I think you need also to add the runtime dependency on `binutils-dev` because of this issue:
- https://gitlab.alpinelinux.org/alpine/aports/issues/10888
- https://github.com/alpinelinux/aports/pull/12008

**Comment by XVilka** [comment]
How I can help to get this merged? Because broken OCaml is so annoying

**Comment by XVilka** [comment]
Any updates on this?

**Comment by XVilka** [comment]
Can be closed because of https://github.com/alpinelinux/aports/commit/d95a53813d93148146b793532b55873416d9af63

**Comment by andyli** [comment]
For the record, the change was reverted in 6f58e167f1865920a5d21859555b905ec1b2cec3.

**Comment by XVilka** [comment]
Note, it is because Camlp4 is being [officially deprecated](https://discuss.ocaml.org/t/the-end-of-camlp4/4216). Most of the packages already got rid of it - either in favor of ppx, menhir/sedlex, or, sadly, camlp5. I made a list in the same post of the packages that still require camlp4, and updated it from time to time. To date most of the commonly installed packages did the migration, thanks to the efforts of the Debian packaging team.

**Comment by XVilka** [comment]
And now 4.08.1 is unusable too: https://gitlab.alpinelinux.org/alpine/aports/issues/11057
